### ISSUE TITLE
Better error handling

### ISSUE THREAD

**Issue by matomesc** [bryanmikaelian/battlenet#7]
Would you be interested in a patch that improves error handling? Right now all callbacks are called with `callback(data)` which does not really follow the node conventions of `callback(err, data)`. I'll be more than happy to submit a pull request to fix this.

**Comment by bryanmikaelian** [comment]
@matomesc Absolutely. My node skills have been falling behind since I am primarily a Rails developer. PR away, I say!
